Search/Metadata/ProviderInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\Metadata;
+
+use Massive\Bundle\SearchBundle\Search\Document;
+
+/**
+ * Provide a more flexible extension point for loading metadata.
```

Comment:
Provide metadata for the search manager from object instances. This allows for more the retrieval of metadata using more than just the class name as a criteria.